test/test_naive_completion.py

```diff
@@ -52,3 +52,21 @@ def test_column_name_completion(completer, complete_event):
         Document(text=text, cursor_position=position),
         complete_event)
     assert result == list(map(Completion, sorted(completer.all_completions)))
+
+def test_various_join_completions(completer, complete_event):
+    for join_type in ['INNER', 'OUTER', 'CROSS', 'LEFT', 'RIGHT', 'FULL']:
+        text = 'SELECT foo FROM bar {join_type} '
```

Comment:
What's the value of `'SELECT foo FROM bar {join_type} '`? should we use f-Strings here?

Reply:
I’m new to Python, so unsure about your comment. I looked up F strings but I don’t see the benefit here. I have the test in a loop so we don’t need a separate test for each keyword.

How would F strings help here?

Reply:
f-strings will replace the variable with its value, but the normal string will do nothing. e.g.:
``` python
>>> for join_type in ['INNER', 'OUTER', 'CROSS', 'LEFT', 'RIGHT', 'FULL']:
...     print('SELECT foo FROM bar {join_type} ')
...
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}

>>> for join_type in ['INNER', 'OUTER', 'CROSS', 'LEFT', 'RIGHT', 'FULL']:
...     print(f'SELECT foo FROM bar {join_type} ')  # f-strings
...
SELECT foo FROM bar INNER
SELECT foo FROM bar OUTER
SELECT foo FROM bar CROSS
SELECT foo FROM bar LEFT
SELECT foo FROM bar RIGHT
SELECT foo FROM bar FULL
```

Reply:
@zzl0 I'm kind of surprised the tests were passing as-is, but I see the mistake now. Thanks for the explanation. I have pushed a fix.

Reply:
> I'm kind of surprised the tests were passing as-is

Could you try to set parameter `smart_completion=True` when calling `get_completions`? this should fail the first version of your tests.